src/types/markers.js

```diff
@@ -48,42 +48,145 @@ export type PaintProfilerMarkerTracing = ProfilerMarkerTracing & {
     | 'Composite',
 };
 
-export type GCMinorMarkerPayload = {
-  type: 'GCMinor',
-  startTime: Milliseconds,
-  endTime: Milliseconds,
-  // nursery is only present in newer profile format.
-  nursery?: {|
-    reason?: string,
-    status?: string,
-  |},
+export type PhaseTime = { [phase: string]: Milliseconds };
+
+export type GCSliceData = {
+  // Slice number within the GCMajor collection.
+  slice: number,
+
+  pause: Milliseconds,
+  when: Milliseconds,
+
+  // The reason for this slice.
+  reason: string,
+
+  // The GC state at the start and end of this slice.
+  initial_state: string,
+  final_state: string,
+
+  // The incremental GC budget for this slice (see pause above).
+  budget: Milliseconds,
+
+  // The number of the GCMajor that this slice belongs to.
+  major_gc_number: number,
+
+  trigger_amount?: number,
+  trigger_threshold?: number,
+
+  // The number of page faults that occured during the slice.
+  page_faults: number,
+
+  start_timestamp: Seconds,
+  end_timestamp: Seconds,
+
+  phase_times: PhaseTime[],
+};
+
+export type GCMajorAborted = {
+  status: 'aborted',
+};
+
+export type GCMajorCompleted = {
+  status: 'completed',
+  // timestamp is present but is usually 0
+  // timestamp: number,
+  max_pause: Milliseconds,
+
+  // The sum of all the slice durations
+  total_time: Milliseconds,
+
+  // The reason from the first slice. see JS::gcreason::Reason
+  reason: string,
```

Comment:
in the future, we'll want an enum here as well, so that we can display user-friendly descriptions of the reason (or at least of some of the reasons)

Reply:
Yes.  It'll help with the threshold fields above which are present for only some reasons.